src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeSummaryRepositoryTest.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.hmcts.reform.blobrouter.data;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.blobrouter.data.model.EnvelopeSummary;
+import uk.gov.hmcts.reform.blobrouter.data.model.EventRecord;
+import uk.gov.hmcts.reform.blobrouter.data.model.NewEnvelope;
+import uk.gov.hmcts.reform.blobrouter.data.model.NewEventRecord;
+import uk.gov.hmcts.reform.blobrouter.data.model.Status;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.UUID;
+
+import static java.time.ZoneOffset.UTC;
+import static java.util.Comparator.comparingLong;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.DELETED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.DELETED_FROM_REJECTED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.DISPATCHED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.DUPLICATE_REJECTED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.FILE_PROCESSING_STARTED;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Event.REJECTED;
+
+@ActiveProfiles({"integration-test", "db-test"})
+@SpringBootTest
+public class EnvelopeSummaryRepositoryTest {
+    @Autowired private EnvelopeRepository envelopeRepository;
+    @Autowired private EventRecordRepository eventRecordRepository;
+    @Autowired private EnvelopeSummaryRepository repo;
+    @Autowired private DbHelper dbHelper;
+
+    private static final String CONTAINER_1 = "container1";
+    private static final String CONTAINER_2 = "container2";
+    private static final String BULKSCAN_CONTAINER = "bulkscan";
+    private static final String FILE_1_1 = "file_name_1_1";
+    private static final String FILE_1_2 = "file_name_1_2";
+    private static final String FILE_1_3 = "file_name_1_3";
+    private static final String FILE_2_1 = "file_name_2_1";
+    private static final String FILE_2_2 = "file_name_2_2";
+    private static final String FILE_2_3 = "file_name_2_3";
+    private static final String FILE_2_4 = "file_name_2_4";
+    private static final String FILE_BULKSCAN_1 = "file_name_bulkscan_1";
+
+    @BeforeEach
+    void setUp() {
+        dbHelper.deleteAll();
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:variabledeclarationusagedistance")
+    void should_find_within_date_range() {
```

Comment:
This test tests quite a lot at once. It would be worth splitting it into multiple tests, each focused on a different thing. For example:
- that the retrieved data maps correctly to fields
- that the search doesn't return records from outside given date range
- that the search doesn't return results for a different container
- that the search doesn't return results for a different file
- event data comes from the last event

Reply:
agree

Reply:
@rkondratowicz @lgonczar 
There are 3 test methods:
- should_find_within_date_range - checks that files before and after given time range are filtered out, also (as well as other methods) it checks that all fields in EventSummary have correct values including container and file name and that event related fields have their values from the last event
- should_skip_bulkscan_repository - checks that the file from 'bulkscan' container is not included in the result
- should_handle_missing_events - checks that in case of no events for a file the result is still successfully fetched; though we expect that every file should have at least 1 event handling this situation (no events for a file) makes the code base bullet proof

I hope this explains details of the test class